staging/src/k8s.io/legacy-cloud-providers/gce/gce_disks.go

```diff
@@ -620,6 +624,66 @@ func (g *Cloud) DisksAreAttached(diskNames []string, nodeName types.NodeName) (m
 	return attached, nil
 }
 
+// BulkDisksAreAttached is a batch function to check if all corresponding disks are attached to the
+// nodes specified with nodeName.
+func (g *Cloud) BulkDisksAreAttached(diskByNodes map[types.NodeName][]string) (map[types.NodeName]map[string]bool, error) {
+	instanceNames := []string{}
+	for nodeName := range diskByNodes {
+		instanceNames = append(instanceNames, mapNodeNameToInstanceName(nodeName))
+	}
+
+	// List all instances with the given instance names
+	// Then for each instance listed, add the disks attached to that instance to a map
+	listedInstances, err := g.getInstancesByNames(instanceNames)
+	listedInstanceNamesToDisks := make(map[string][]*compute.AttachedDisk)
+	for _, instance := range listedInstances {
+		listedInstanceNamesToDisks[instance.Name] = instance.Disks
+	}
+	if err != nil {
+		if err == cloudprovider.InstanceNotFound {
+			notFound := []string{}
+			for _, instanceName := range instanceNames {
+				if _, found := listedInstanceNamesToDisks[instanceName]; !found {
+					notFound = append(notFound, instanceName)
+				}
+			}
+			klog.Warningf("Failed to find some instances: %v", notFound)
+		} else {
+			return nil, fmt.Errorf("error listing instances: %v", err)
+		}
+	}
+
+	verifyDisksAttached := make(map[types.NodeName]map[string]bool)
+
+	// For each node and its desired attached disks that needs to be verified
+	for nodeName, attachedDisks := range diskByNodes {
+		verifyDisksAttachedToNode := make(map[string]bool)
+
+		if disksActuallyAttached, ok := listedInstanceNamesToDisks[mapNodeNameToInstanceName(nodeName)]; ok {
```

Comment:
can you opaque this into a separate function (this one is already quite long)
so sth like:
```
verifyDisksAttached[nodeName] = verifyDisksAttachedToNode(attachedDisks, disksActuallyAttached)
```